openedx_tagging/core/tagging/system_defined_taxonomies/object_tags.py

```diff
@@ -0,0 +1,210 @@
+    """
+    LanguageTaxonomy = 1
+
+
+class SystemDefinedObjectTagMixin:
+    """
+    Mixing for ObjectTags used on all system defined taxonomies
+
+    `system_defined_taxonomy_id``is used to connect the 
+    ObjectTag with the system defined taxonomy.
+    This is because there can be several ObjectTags
+    for the same Taxonomy, ex:
+    
+    - LanguageCourseObjectTag
+    - LanguageBlockObjectTag
+
+    On the example, there are ObjectTags for the same Language
+    taxonomy but with different objects.
+
+    Using this approach makes the connection between the ObjectTag
+    and system defined taxonomy as hardcoded and can't be changed.
+    """
+
+    system_defined_taxonomy_id = None
+
+    def _check_system_taxonomy(self, taxonomy: Taxonomy = None):
+        """
+        Validates if the taxonomy is system-defined and match
+        with the name stored in the object tag
+        """
+        return (
+            bool(taxonomy) and
+            taxonomy.system_defined and
+            taxonomy.id == self.system_defined_taxonomy_id
+        )
+
+
+class OpenSystemObjectTag(OpenObjectTag, SystemDefinedObjectTagMixin):
+    """
+    Free-text object tag used on system-defined taxonomies
+    """
+
+    class Meta:
+        proxy = True
+
+    def _check_taxonomy(self):
+        return (
+            super()._check_taxonomy() and
+            self._check_system_taxonomy(self.taxonomy)
+        )
+
+    
+class ClosedSystemObjectTag(ClosedObjectTag, SystemDefinedObjectTagMixin):
+    """
+    Object tags linked to a closed system-taxonomy
+    """
+
+    class Meta:
+        proxy = True
+
+    def _check_taxonomy(self):
+        return (
+            super()._check_taxonomy() and
+            self._check_system_taxonomy(self.taxonomy)
+        )
+
+
+class ModelObjectTag(OpenSystemObjectTag):
+    """
+    Object tags used with tags that relate to the id of a model
+    
+    This object tag class is not registered as it needs to have an associated model
+    """
+
+    class Meta:
+        proxy = True
+
+    tag_class_model = None
+
+
+    def _check_taxonomy(self):
+        """
+        Validates if has an associated Django model that has an Id
+        """
+        if not super()._check_taxonomy():
+            return False
+    
+        if not self.tag_class_model:
+            return False
+
+        # Verify that is a Django model
+        if not issubclass(self.tag_class_model, models.Model):
+            return False
+
+        # Verify that the model has 'id' field
+        try:
+            self.tag_class_model._meta.get_field('id')
+        except FieldDoesNotExist:
+            return False
+
+        return True
+    
+    def _check_instance(self):
+        """
+        Validates if the instance exists
+        """
+        try:
+            intance_id = int(self.value)
+        except ValueError:
+            return False
+        return self.tag_class_model.objects.filter(id=intance_id).exists()
```

Comment:
As noted, we can't rely on the primary key always being called `id`, and we don't know whether it's an integer.

So I think this should be:
```suggestion
        return self.tag_class_model.objects.filter(pk=self.value).exists()
```